### ISSUE TITLE
Feature Request: `AWS::Lambda::LayerVersion` - Verify `CompatibleRuntimes`

### ISSUE THREAD

**Issue by r-heimann** [aws-cloudformation/cfn-lint#4427]
### Is this feature request related to a new rule or cfn-lint capabilities?

New capability

### Describe the feature you'd like to request

Currently cfn-lint is unable to verify LambdaLayers `CompatibleRuntimes`:

```yaml
AWSTemplateFormatVersion: 2010-09-09
Description: Test


Resources:
  Layer:
    Type: AWS::Lambda::LayerVersion
    Properties:
      CompatibleRuntimes:
        - python3.16 # <--------------
      Content:
        S3Bucket: 01234567890-test
        S3Key: "test.zip"
```

Error:

>1 validation error detected: Value '[null]' at 'compatibleRuntimes' failed to satisfy constraint: Member must satisfy enum value set: [nodejs20.x, python3.14, provided.al2023, python3.12, python3.13, nodejs24.x, nodejs22.x, java25, python3.10, python3.11, java21, ruby3.3, ruby3.4, ruby3.2, python3.8, python3.9, java17, nodejs16.x, dotnet10, dotnet8, java11, dotnet6, nodejs18.x, provided.al2, java8.al2

### Describe the solution you'd like

A check which verifies if the value is correct and usable.

### Additional context

\-

### Is this something that you'd be interested in working on?

- [ ] 👋 I may be able to implement this feature request

### Would this feature include a breaking change?

- [ ] ⚠️ This feature might incur a breaking change
